### ISSUE TITLE
unsure if reported already

### ISSUE THREAD

**Issue by falcon470** [net-lisias-ksp/KSP-Recall#63]
so i was trying to make a rocket lanchy from a plane merge assembly.... well didn go wsell took apart my whole rocket, checked here and found the new version instead of taking apart my whole rocket it only shifted the first and second stages a little off from each other its close enough to fix but not perfect yet i havent downloaded the new tweak scale version yet, so mabey thats it any way ill post a screen shot, wish id have taken a picture of it before but needless to say that top section was all dismanted solar panels 10 feet above craft 
same with hte parachute and other parts, i am playing heavily modded with realism overhaul rss as the base so i am not entirely
certain if this is just my install but still good work, that made things at least livable
![20230327_112940](https://user-images.githubusercontent.com/59348243/227989248-4e54fd71-558a-411b-bcae-1e1a066b6f1f.jpg)

**Comment by Lisias** [comment]
hi, @falcon470 !

Oh, boy… This one is a messy - and pretty annoying - hard nut to crack.

### TL;DR:

It's fixed when using KSP-Recall 0.4.0.0 with TweakScale Beta 2.5.0.59 (current latest).

There're currently  two possible solutions:

* Using SubAssemblies instead of Craft Merge.
    + SubAssemblies are working fine, it's Merge the Editor's problem that I failed to fix on TS 2.4.x series. 
* Risking your arse using TS Beta 2.5.0.59 (or later).
    + The Beta is almost Release Candidate by now, but since I got my free time **royally** screwed this year by 3rd parties, I didn't fully tested the damn thing as I would like
        - so, there's a chance that something new my bite your arse using it.

### Gory Details

This is another KSP's Editor screw up that I misdiagnosed in the past, and ended up coding Variant support on TweakScale over the shoddy bug. Then things started to get messy on other parts, and I misdiagnosed the problem again and coded a "fix" on KSP-Recall (sigh).

Well, this is what really had happened in the past and how I fixed it: over the years, the KSP Editor is introducing bugs probably related to shoddy workarounds when implementing support for Variants on it. This ended up creating different misbehavirous, depending if you are creating a craft, loading it, merging it or applying a sub-assembly on it. Different KSP versions introduced different bugs on Editor, being the initial problematic release KSP 1.4.3 or, more probably, 1.4.2 (The Brief).

Some of these misbehaviours I let leak into TweakScale on the 2.4.x series. And, then, when things started to misbehave on non Variant parts, I had (wrongly) concluded that it was another bug on Editor - without realising that this time **I** was the source of the misbehaviour because I had "fixed" the wrong problem on the wrong place.

**NOW** KSP-Recall is (hopefully) doing the right thing, and the problem is solely on TweakScale 2.4.7.x (where I fixed a lot of small bugs, but not all of them).

TweakScale 2.5 is a major overhaul of the whole Variant thingy, where a lot of structural problems (and fundamental misdiagnoses) were tackled down (and not only on the Variant support!). Problem: besides being a huge change in need to be correctly validated, this made some Companions incompatible, so I could not push TS 2.5 into the mainstream until early this year, when I finally could check and synchronise things on everybody (this is a very time consuming task).

I had planned to start pushing the TS Beta into Mainstream this year's late January, late February at worst, but problems on pushing them into CKAN royally screwed up my scheduling. Now it's March, when my Country really start to go back to work economically (nothing really works until Carnival around here), and now my available time to do such huge and critical publishings are essentially gone and limited to a fraction of my WeekEnds only.

And so I decided to push a stop hole called 2.4.7.x instead, where the "safe" fixes were merged from Beta - and the remaining bugs are at least known and have workarounds - instead of risking pushing 2.5 without all the needed testings and discovering a unexpected and nasty bug bitting user's arses.

The TS 2.5 Variant support was reworked from the bottom, and so I'm still charting the new seas. I didn't checked for regressions over all the (now considerable big) closed issues.

Sorry for that - I was my firm intention to have 2.5 published as mainstream by now, but… (sigh²).

Cheers!

**Comment by Lisias** [comment]
I may had misdiagnosed this one. See $65 .
